kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala

```diff
@@ -445,20 +445,32 @@ object KubernetesApplicationOperation extends Logging {
           .find(cs => appStateContainer.equalsIgnoreCase(cs.getName))
       case KubernetesApplicationStateSource.POD => None
     }
-    val applicationState = containerStatusToBuildAppState
+
+    val podAppState = podStateToApplicationState(pod.getStatus.getPhase)
+    val containerAppState = containerStatusToBuildAppState
```

Comment:
changes look good to me, can we leave some comments here with an example you listed in the PR description? also, please update the conf docs of `kyuubi.kubernetes.application.state.source`, to match the updated check logic

Reply:
done